test/e2e/autoscale_test.go

```diff
@@ -357,12 +383,90 @@ func assertAutoscaleUpToNumPods(ctx *testContext, curPods, targetPods float64, d
 	}
 }
 
+func assertGracefulScaledown(t *testing.T, ctx *testContext, size int) error {
+	// start x running pods; x == size
+	hostConnMap, err := uniqueHostConnections(t, ctx.names, size)
+	if err != nil {
+		return err
+	}
+
+	// only keep openConnCount connections open for the test
+	openConnCount := size / 2
+	deleteHostConnections(hostConnMap, size-openConnCount)
+
+	defer deleteHostConnections(hostConnMap, openConnCount)
+	grp := errgroup.Group{}
+	grp.Go(func() error {
+		timer := time.NewTicker(2 * time.Second)
+		for _ = range timer.C {
+			readyCount, err := numberOfReadyPods(ctx)
+			if err != nil {
+				return err
+			}
+
+			if int(readyCount) < openConnCount {
+				return fmt.Errorf("Failed keeping the right number of pods. Ready(%d) != Expected(%d)", int(readyCount), openConnCount)
+			}
+
+			if int(readyCount) == openConnCount {
+				pods, err := allPods(ctx)
+				if err != nil {
+					return err
+				}
+
+				for _, p := range pods {
+					if p.Status.Phase != corev1.PodRunning || p.DeletionTimestamp != nil {
+						continue
+					}
+
+					if _, ok := hostConnMap.Load(p.Name); !ok {
+						return fmt.Errorf("Failed by keeping the wrong pod %s", p.Name)
```

Comment:
```suggestion
						return fmt.Errorf("failed by keeping the wrong pod %s", p.Name)
```

Reply:
haha between you and @vagababov it is hard to tell if you like messages whether or not to start with capital letters 😄 ... I think he suggested the capital letters.

Reply:
It's very easy:
`t.XF()` -- always capital
`fmt.Errorf/errors.New` -- always lowercase :)

Reply:
hmmm ok. good to know. is this a knative convention? or some other pattern?

Reply:
golang.

Reply:
how about `logger.XF()`?

Reply:
if you return `error` it's lowercase, else: full sentences
https://github.com/golang/go/wiki/CodeReviewComments#error-strings